src/loader/mod.rs

```diff
@@ -39,110 +33,61 @@ use vm_memory::{Address, Bytes, GuestAddress, GuestMemory, GuestUsize};
 #[cfg_attr(feature = "cargo-clippy", allow(clippy::all))]
 pub mod bootparam;
 
-#[cfg(feature = "elf")]
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-#[allow(missing_docs)]
-#[cfg_attr(feature = "cargo-clippy", allow(clippy::all))]
-pub mod start_info;
+mod struct_util;
 
-#[allow(dead_code)]
-#[allow(non_camel_case_types)]
-#[allow(non_snake_case)]
-#[allow(non_upper_case_globals)]
-#[cfg_attr(feature = "cargo-clippy", allow(clippy::all))]
-mod elf;
-#[cfg(any(feature = "elf", feature = "bzimage"))]
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-mod struct_util;
+mod x86_64;
+
+#[cfg(target_arch = "aarch64")]
+mod aarch64;
+
+#[cfg(all(feature = "bzimage", any(target_arch = "x86", target_arch = "x86_64")))]
+pub use x86_64::bzimage::BzImage;
+#[cfg(all(feature = "bzimage", any(target_arch = "x86", target_arch = "x86_64")))]
+pub use x86_64::bzimage::Error as BzImageError;
+
+#[cfg(all(feature = "elf", any(target_arch = "x86", target_arch = "x86_64")))]
+pub use x86_64::elf::Elf;
+#[cfg(all(feature = "elf", any(target_arch = "x86", target_arch = "x86_64")))]
+pub use x86_64::elf::Error as ElfError;
```

Comment:
I think you also need a:
```Rust
#[cfg(all(feature = "elf", any(target_arch = "x86", target_arch = "x86_64")))]
pub use x86_64::elf::start_info;
```

Reply:
I think (hope) both this issue and @andreeaflorescu's nit https://github.com/rust-vmm/linux-loader/pull/24#issuecomment-600578026 are solved in the latest commit. The `elf` and `bzimage` modules are `use`d as-is, simplifying the imports in `loader/mod.rs`, at the expense of the user prepending `elf::` and `bzimage::` to structs and calls.